### PR TITLE
[FEATURE] BatchDefinition.get_batch_identifiers_list

### PR DESCRIPTION
**Pull request by tyler-hoffman**
This method is a follow up to https://github.com/great-expectations/great_expectations/pull/10295. Some users previously used `get_batch_list_from_batch_request` to see the available batches, but that was replaced with private methods. This PR exposes a public method on `BatchDefinition` to get the available batch identifiers list. Unlike the previous method, this will not read in all data, and is more performant.

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/integration/common_workflows/test_sql_asset_workflows.py [hunk 1]
```diff
@@ -109,6 +109,55 @@ def _create_test_cases():
     ]
 
 
+def test_get_batch_identifiers_list__simple(
```
Comment (tyler-hoffman) [inline tests/integration/common_workflows/test_sql_asset_workflows.py:112]: This test requires the DB to be seeded already; TODO: Fix in another PR.

Comment (cdkini) [reply tests/integration/common_workflows/test_sql_asset_workflows.py:112]: Can we leave a comment about this?

File: tests/integration/common_workflows/test_sql_asset_workflows.py [hunk 2]
```diff
@@ -109,6 +109,55 @@ def _create_test_cases():
...
+    postgres_daily_batch_definition: BatchDefinition,
+) -> None:
+    batch_definition = postgres_daily_batch_definition
+    batch_identifiers_list = batch_definition.get_batch_identifiers_list()
+
+    assert len(batch_identifiers_list) == 10
+    # just spot check the edges
+    assert batch_identifiers_list[0] == {"year": None, "month": None, "day": None}
+    assert batch_identifiers_list[-1] == {"year": 2001, "month": 1, "day": 1}
```
Comment (cdkini) [inline tests/integration/common_workflows/test_sql_asset_workflows.py:121]: Why are the values `None` for this first one?

File: tests/integration/common_workflows/test_filesystem_asset_workflows.py [hunk 3]
```diff
@@ -185,6 +185,115 @@ def _create_test_cases():
...
+    ],
+)
+def test_get_batch_identifiers_list__respects_order(
+    batch_definition_fixture_name: str,
+    request: pytest.FixtureRequest,
+) -> None:
+    batch_definition: BatchDefinition = request.getfixturevalue(batch_definition_fixture_name)
+    batch_identifiers_list = batch_definition.get_batch_identifiers_list()
+
+    assert len(batch_identifiers_list) == 36
```
Comment (cdkini) [inline tests/integration/common_workflows/test_filesystem_asset_workflows.py:234]: For all of these numbers when we assert length, should we add a comment about what that number is and why it makes sense?

**Review by cdkini — APPROVED** [review]

File: tests/integration/common_workflows/test_sql_asset_workflows.py (hunk 2 above)
Comment (tyler-hoffman) [reply tests/integration/common_workflows/test_sql_asset_workflows.py:121]: There was one entry in the sample data where those fields are None, and when values are missing, they come first when sorting ascending.

**Pull request merged**